src/services/lnd/onchain-service.ts

```diff
@@ -27,26 +28,42 @@ export const OnChainService = (
     return new OnChainServiceUnavailableError(err)
   }
 
-  const getIncomingTransactions = async (
+  const listTransactions = async (
     scanDepth: number,
-  ): Promise<SubmittedTransaction[] | OnChainServiceError> => {
+  ): Promise<GetChainTransactionsResult | OnChainServiceError> => {
     try {
       const { current_block_height } = await getHeight({ lnd })
 
       // this is necessary for tests, otherwise after may be negative
       const after = Math.max(0, current_block_height - scanDepth)
 
-      const result = await getChainTransactions({
+      return getChainTransactions({
         lnd,
         after,
       })
-
-      return extractIncomingTransactions(decoder, result)
     } catch (err) {
       return new UnknownOnChainServiceError(err)
     }
   }
 
+  const listIncomingTransactions = async (
+    scanDepth: number,
+  ): Promise<SubmittedTransaction[] | OnChainServiceError> => {
+    const txs = await listTransactions(scanDepth)
+    if (txs instanceof Error) return txs
+
+    return extractIncomingTransactions({ decoder, txs })
+  }
+
+  const listOutgoingTransactions = async (
+    scanDepth: number,
```

Comment:
this could be its own type

Reply:
True - when there is a primitive type that naturally describes the value I don't mind using the primitive type. Here scan depth really is a number. But I guess we could add additional constraints (`x > 0 && < maxDepth`) which would merit the type.

Reply:
done https://github.com/GaloyMoney/galoy/commit/f308efa7545cfab98b612b82ecedb06c0ada099f